Yoast/YoastSEO.js
# Issue1249 typescript jest test framework

## Summary

This PR can be summarized in the following changelog entry:
Changed testing framework to Jest.

## Relevant technical choices:

* Changed testing framework to Jest.

## Test instructions

This PR can be tested by following these steps:

* Check if the existing tests still work and change a few tests to see if the tests fail.
* Create sum.ts in the /src folder and insert the code given below.
`export function sum(a: number, b: number): number {
    return a + b;
}`

And create a sumSpec.ts file in the /spec folder with the following test:
```
const sum = require('../src/sum');

it('adds 1 + 2 to equal 3 in TScript', ()=> {
    expect(sum(1, 2)).toBe(3);
});
```


Fixes #1249
